charmcraft/extensions/gunicorn.py

```diff
@@ -223,6 +223,12 @@ def is_experimental(base: tuple[str, ...] | None) -> bool:  # noqa: ARG004
         """Check if the extension is in an experimental state."""
         return False
 
+    @override
+    def get_parts_snippet(self) -> dict[str, Any]:
+        """Return the parts to add to parts."""
+        # rust is needed to build pydantic-core, a dependency of flask.
+        return {"flask-framework/rust-deps": {"plugin": "nil", "build-packages": ["cargo"]}}
```

Comment:
I think build-packages get installed before any steps are executed, so I guess this doesn't need to worry about other parts defining `after: [flask-framework/rust-deps]`?

Reply:
Correct :-)